pkg/kubevisor/visor.go

```diff
@@ -153,12 +172,37 @@ func (k *Visor) DeleteResource(ctx context.Context, t ResourceType, name string)
 		return errors.Errorf("unknown Kubernetes resource type provided for deletion: '%s'", t)
 	}
 
+	//For jobs we also need to delete any associated pods
+	if t == ResourceTypeJobs {
+		//Retrieve job UUID to look up associated pods
```

Comment:
I would suggest moving this to the service package, this package also contains a pod transformer already. @LilliJane  knows more about this, ask here if you have questions!